addons/html_editor/static/src/utils/base_container.js

```diff
@@ -0,0 +1,54 @@
+export const BASE_CONTAINER_CLASS = "o-paragraph";
+
+export const SUPPORTED_BASE_CONTAINER_NAMES = ["P", "DIV"];
+
+export class BaseContainerFactory {
+    static selector;
+
+    constructor(nodeName, document = null) {
+        if (!document && window) {
+            document = window.document;
+        }
+        this.document = document;
+        // Default technical value is "P" (for tests and semantics).
+        // See "baseContainer" html_field config property for the functional default.
+        this.nodeName =
+            nodeName && SUPPORTED_BASE_CONTAINER_NAMES.includes(nodeName) ? nodeName : "P";
+        this.classSet = new Set();
+        if (this.nodeName !== "P") {
+            this.classSet.add(BASE_CONTAINER_CLASS);
+        }
+    }
+
+    get class() {
+        return [...this.classSet].join(" ");
+    }
+
+    get selector() {
+        return `${this.nodeName}${this.classSet.size ? "." : ""}${[...this.classSet].join(".")}`;
+    }
+
+    get tagName() {
```

Comment:
It's funny because this is called _tag_ name but it returns the value of a variable called _node_ name... ^^' Though to be fair I think we are inconsistent about that as well in the editor.